presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/MergeWindows.java

```diff
@@ -85,30 +85,45 @@ protected PlanNode visitPlan(
                 RewriteContext<Multimap<WindowNode.Specification, WindowNode>> context)
```

Comment:
I see, your solution would be better in some cases than the one I provided here but it's also more complicated for negligible gain I'd say (it's still not optimal). I don't think it's worth complicating the optimizer for rare cases which never occurred in our use cases, nor benchmarks at our side.